posthog/queries/retention/retention_events_query.py

```diff
@@ -191,17 +189,20 @@ def get_timestamp_field(self) -> str:
             return f"{self._trunc_func}(toTimeZone(toDateTime({self.EVENT_TABLE_ALIAS}.timestamp, 'UTC'), %(timezone)s)) AS event_date"
 
     def _determine_should_join_distinct_ids(self) -> None:
-        if (
+        non_person_id_aggregation = (
             self._filter.aggregation_group_type_index is not None or self._aggregate_users_by_distinct_id
-        ) and not self._column_optimizer.is_using_cohort_propertes:
+        )
+        is_using_cohort_propertes = self._column_optimizer.is_using_cohort_propertes
+        if self._person_on_events_mode == PersonOnEventsMode.V1_ENABLED or (
```

Comment:
shouldn't this be the case for PoEv2 as well (we don't want to join distinct_ids then either, do we?)